AmplifyPlugins/DataStore/AWSDataStoreCategoryPluginAuthIntegrationTests/AWSDataStoreCategoryPluginAuthIntegrationTests.swift

```diff
@@ -0,0 +1,230 @@
+            signOut()
+        }
+    }
+
+    override func tearDown() {
+        signOut()
+        Amplify.reset()
+    }
+
+    /// A user can persist data in the local store without signing in. Once the user signs in,
+    /// the sync engine will start and sync the mutations to the cloud. Once the reconcillation is complete, retrieving
+    /// the same data will contain ownerId
+    ///
+    /// - Given: A DataStore plugin configured with SocialNote model containing with auth rules
+    /// - When:
+    ///    - User is not signed in, then user can successfully save a note to local store
+    ///    - User remains signed out, then user can successfully retrieve the saved note, with empty owner field
+    ///    - User signs in, retrieves note, sync engine is started and reconciles local store with the ownerId
+    ///    - The note now it contains the ownerId
+    func testUnauthenticatedSavesToLocalStoreIsReconciledWithCloudStoreAfterAuthentication() throws {
+        let savedLocalNote = saveNote(content: "owner saved note")
+        let queriedNoteOptional = queryNote(byId: savedLocalNote.id)
+        guard let note = queriedNoteOptional else {
+            XCTFail("Faled to query local note")
+            return
+        }
+        XCTAssertNil(note.owner)
+
+        let syncReceivedInvoked = expectation(description: "Received SyncReceived event")
+        var remoteNoteOptional: SocialNote?
+        let syncReceivedListener = Amplify.Hub.listen(to: .dataStore, eventName: syncReceived) { payload in
+            guard let mutationEvent = payload.data as? MutationEvent,
+                let note = try? mutationEvent.decodeModel() as? SocialNote else {
+                    XCTFail("Can't cast payload as mutation event")
+                    return
+            }
+            if note.id == savedLocalNote.id {
+                remoteNoteOptional = note
+                syncReceivedInvoked.fulfill()
+            }
+        }
+        guard try HubListenerTestUtilities.waitForListener(with: syncReceivedListener, timeout: 5.0) else {
+            XCTFail("syncReceivedListener registered for hub")
+            return
+        }
+
+        signIn(username: user1.username, password: user1.password)
+
+        wait(for: [syncReceivedInvoked], timeout: TestCommonConstants.networkTimeout)
+        Amplify.Hub.removeListener(syncReceivedListener)
+        guard let remoteNote = remoteNoteOptional else {
+            XCTFail("Should have received a SyncReceived event with the remote note reconciled to local store")
+            return
+        }
+        XCTAssertNotNil(remoteNote.owner)
+    }
+
+    /// A signed in user (the owner) creates some data in local store will be synced to cloud. After `DataStore.clear`,
+    /// the data can no longer be retrieved. Signing back in with another user will update the local store with all
+    /// the data that can be read by that the user in the sync process. Then other user can read the owner's data.
```

Comment:
User1, while signed in, creates some data, called "NewUser1Data" in a local store.  We wait for this data to be synced to the cloud and then call DataStore.clear() so that data will be destroyed in the local data store and the remote sync engine will be halted.  After this finishes, we sign out of user1 and sign in with user2, which will restart the sync engine and read all of data from the backend including "NewUser1Data".

Reply:
thanks for the improved wording